src/js/server/schemaMiddleware.js

```diff
@@ -0,0 +1,186 @@
+/*
+
+    "Gatekeeper" middleware that rejects any request whose JSON payloads are not valid. See this component's
+    documentation for more details:
+
+    https://github.com/the-t-in-rtf/gpii-json-schema/blob/GPII-1336/docs/middleware.md
+
+ */
+"use strict";
+var fluid = require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+
+require("./schemaHandler");
+
+fluid.defaults("gpii.schema.middleware.handler", {
+    gradeNames: ["gpii.schema.handler"],
+    invokers: {
+        handleRequest: {
+            func: "{that}.sendResponse",
+            args: ["{that}.options.statusCode", "{that}.options.message"]
+        }
+    }
+});
+
+fluid.registerNamespace("gpii.schema.middleware");
+
+gpii.schema.middleware.rejectOrForward  = function (that, req, res, next) {
```

Comment:
See above comment - to be helpful to users, this should actually conform to the middleware contract